inc/saiswitch.h

```diff
@@ -1166,6 +1166,20 @@ typedef enum _sai_switch_attr_t
      */
     SAI_SWITCH_ATTR_ECMP_DEFAULT_HASH_SEED,
 
+    /**
+     * @brief SAI ECMP default hash offset
+     *
+     * When set, the output of the ECMP hash calculation will be shifted or
+     * rotated by the specified number of bits. The direction and exact behavior
+     * (shift vs offset) as well as the maximum rotation supported may be
+     * ASIC-specific.
```

Comment:
This flexibility in the vendor implementation makes it hard to use SAI implementations from different vendors. Why can't we be more prescriptive? 

If there are truly vendors that implements this as a shift vs rotation, can we have two fields SHIFT_OFFSET and ROTATION_OFFSET and an API to query which one is supported? 

Rotation direction shouldn't matter how it's implemented in hardware, because a SAI implementation can always do field_width - shift_length to flip the rotation.

Reply:
We discussed this a bit in the community meeting, and there was some reluctance to specify the exact ASIC behavior. I think there are probably some good reasons to specify the behavior here (particularly to prevent polarization in mixed-ASIC networks). I'd personally be on board with defining this as "rotate right by N bits" but would need the ASIC vendors to weigh in. @JaiOCP @marian-pritsak @ashutosh-agrawal @lguohan any comments?

Reply:
I agree with @konne88 here. We can define the exact behavior for the "offset" attribute here and other implementations like "shift" etc can be added later as additional attributes.

Reply:
Updated to specify right rotation.